### PR TITLE
Adapt window to fit screen and fix flakes

### LINKED ISSUE #3: window does not resize according to screen size
No text is centrally aligned and the circles in the game overflows the screen.

Reproduce: Just run the activity.

Tested on SaoS on two different screens.

### PR DESCRIPTION
**Pull request by Dimi20cen**
Fixes #3 

@chimosky @sourabhaa

### CODE REVIEW

**Comment by Dimi20cen** [comment]
Hey, is that okay to merge? @chimosky @sourabhaa

File: sprites.py [hunk 1]
```diff
@@ -145,23 +145,23 @@ def __init__(self, game):
         MovingSprite.__init__(self, game)
         
         self.speed = 15
-        self.image = pygame.Surface((40,40))
+        self.image = pygame.Surface((int(40 * self.game.scale), int(40 * self.game.scale)))
         self.image.fill(white)
-        pygame.draw.circle(self.image, black, (20,20), 20)
-        pygame.draw.circle(self.image, red, (20,20), 17)
+        pygame.draw.circle(self.image, black, (int(20 * self.game.scale), int(20 * self.game.scale)), int(20 * self.game.scale))
+        pygame.draw.circle(self.image, red, (int(20 * self.game.scale), int(20 * self.game.scale)), int(17 * self.game.scale))
```
Comment (chimosky) [inline sprites.py]: One of those places where breaking up the args to multi line would help make it more readable and be pep8 compliant.

**Review by chimosky — CHANGES_REQUESTED** [review]

File: cells.py [hunk 2]
```diff
@@ -25,25 +25,26 @@
...
 import cursor
 
 from gettext import gettext as _
 
-################################################################################
+###############################################################################
 # Game
-################################################################################
+###############################################################################
+
```
Comment (chimosky) [inline cells.py]: These lines can be removed entirely as they have no use.

File: cells.py [hunk 3]
```diff
@@ -53,17 +54,19 @@ def load_all(self):
         if new_height <= self.ScreenHeight:
             self.ScreenHeight = new_height
         else:
-            self.ScreenWidth = int(self.ScreenHeight * (aspect_width / aspect_height))
-        self.scale = self.ScreenWidth / 1200  # 1200x900(4/3) is the base resolution
+            aspect_ratio = aspect_width / aspect_height
+            self.ScreenWidth = int(self.ScreenHeight * (aspect_ratio))
+        self.scale = self.ScreenWidth / 1200
```
Comment (chimosky) [inline cells.py:55]: Does floor division affect the scale when used here?

File: cells.py [hunk 4]
```diff
@@ -53,17 +54,19 @@ def load_all(self):
...
 
         self.screen = pygame.display.get_surface()
-        if not(self.screen):
-            self.screen = pygame.display.set_mode((self.ScreenWidth, 
-                                        self.ScreenHeight), pygame.FULLSCREEN)
+        if not (self.screen):
+            self.screen = pygame.display.set_mode((self.ScreenWidth,
+                                                   self.ScreenHeight),
+                                                  pygame.FULLSCREEN)
```
Comment (chimosky) [inline cells.py]:
```python3
self.screen = pygame.display.set_mode(
                  (self.ScreenWidth,  self.ScreenHeight),
                  pygame.FULLSCREEN)
```

What'd you think?

File: cells.py [hunk 5]
```diff
@@ -290,11 +296,12 @@ def help(self):
                 if event.type == pygame.QUIT:
                     self.running = False
                 # and this one is for the "ESC" key
-                if event.type == pygame.KEYDOWN and event.key == pygame.K_ESCAPE:
-                        helping = False
+                if (event.type == pygame.KEYDOWN
+                    and event.key == pygame.K_ESCAPE):
+                    helping = False
```
Comment (chimosky) [inline cells.py]: The indentation here should make it obvious where the condition ends and that's difficult to tell at first glance.

File: colors.py [hunk 6]
```diff
@@ -14,13 +14,14 @@
...
 # Colors
-################################################################################
-red = (255,0,0)
-green = (0,255,0)
-blue = (0,0,255)
-yellow = (255,255,0)
-black = (0,0,0)
-grey = (150,150,150)
-white = (255,255,255)
+###############################################################################
```
Comment (chimosky) [inline colors.py]: There's no need for this.

File: pieces.py [hunk 7]
```diff
@@ -18,42 +18,47 @@
 from sprites import Group, Text, Prisoner
 from colors import red, green, black, white
 
-################################################################################
+###############################################################################
 # The Cells
-################################################################################
+###############################################################################
+
```
Comment (chimosky) [inline pieces.py]: This too.

File: pieces.py [hunk 8]
```diff
@@ -70,34 +75,36 @@ def setAdjHS(self, hs):
...
             self.getMyHS().addPrisoner(prisoner)
 
-################################################################################
+###############################################################################
 # The Escape Area
-################################################################################
-        
+###############################################################################
+
+
```
Comment (chimosky) [inline pieces.py]: This too.

File: pieces.py [hunk 9]
```diff
@@ -70,34 +75,36 @@ def setAdjHS(self, hs):
...
     def addPrisoner(self, prisoner):
         self.prisoners.add(prisoner)
 
-################################################################################
+###############################################################################
 # The Hideout
-################################################################################
-           
+###############################################################################
```
Comment (chimosky) [inline pieces.py]: This and every other one like this.

File: cells.py [hunk 10]
```diff
@@ -173,31 +176,31 @@ def setupBoard(self):
...
+                         color=black)
             text1.rect.center = (int(600 * self.scale), int(450 * self.scale))
-            text2 = Text(_("(h)elp"), size = int(50 * self.scale))
+            text2 = Text(_("(h)elp"), size=int(50 * self.scale))
             text2.rect.topleft = (int(10 * self.scale), int(10 * self.scale))
             self.text = Group((text1, text2))
 
             if (len(self.escArea.prisoners.sprites()) == self.cell_count
-                and self.guard.moving == False):
+               and not self.guard.moving):
```
Comment (chimosky) [inline cells.py]: The indentation here looks wrong.

File: sprites.py [hunk 11]
```diff
@@ -75,39 +79,49 @@ def __init__(self, game, species, x, y, LorR):
...
     def update(self):
         m = pygame.mouse.get_pos()
         d = self.game.dist(m[0], m[1], self.rect.centerx, self.rect.centery)
         button1, button2, button3 = pygame.mouse.get_pressed()
         a = self.game.escArea.prisoners
         if ((d <= int(20 * self.game.scale)) and
-            self in a.sprites() and 
-            not self.game.guard.moving):
+            self in a.sprites() and not
+           self.game.guard.moving):
```
Comment (chimosky) [inline sprites.py]: The old method is better as it's more readable, shows the variable the not is being applied on where it's used.

File: sprites.py (hunk 1 above)
Comment (Dimi20cen) [reply sprites.py]: Yep, that's already fixed in the fix flake8 commit, [fc3d2a1](https://github.com/sugarlabs/cellgame/pull/12/commits/fc3d2a1d0e489015cc2fa74d7525ea98644c73ec)

File: pieces.py (hunk 8 above)
Comment (Dimi20cen) [reply pieces.py]: Fixed

File: pieces.py (hunk 9 above)
Comment (Dimi20cen) [reply pieces.py]: Fixed

File: cells.py (hunk 3 above)
Comment (Dimi20cen) [reply cells.py:55]: Using floor division would affect the scale but the effect would be minimal

File: cells.py (hunk 4 above)
Comment (Dimi20cen) [reply cells.py]: Yep, seems better, fixed

File: cells.py (hunk 5 above)
Comment (Dimi20cen) [reply cells.py]: Yep and a flake8 was present, fixed

File: cells.py (hunk 10 above)
Comment (Dimi20cen) [reply cells.py]: Fixed

File: sprites.py (hunk 11 above)
Comment (Dimi20cen) [reply sprites.py]: Agree, reverted

**Comment by Dimi20cen** [comment]
I rebased to master to make merge possible

File: sprites.py [hunk 12]
```diff
@@ -75,37 +75,37 @@ def __init__(self, game, species, x, y, LorR):
         self.LorR = LorR
 
         # this is the image you see most of the time
-        self.original = pygame.Surface((40,40)).convert()
+        self.original = pygame.Surface((int(40 * self.game.scale), int(40 * self.game.scale))).convert()
         self.original.fill(yellow)
-        pygame.draw.circle(self.original, black, (20,20), 20)
+        pygame.draw.circle(self.original, black, (int(20 * self.game.scale), int(20 * self.game.scale)), int(20 * self.game.scale))
```
Comment (chimosky) [inline sprites.py]: Could you break long lines like these to be pep8 compliant?

**Review by chimosky — CHANGES_REQUESTED** [review]

**Comment by chimosky** [comment]
Again, please leave flake8 fixes to a different PR. Makes it easier to review.

File: cells.py [hunk 13]
```diff
@@ -30,9 +30,7 @@
 
 from gettext import gettext as _
 
-###############################################################################
 # Game
```
Comment (chimosky) [inline cells.py]: These are supposed to be removed too as the code makes it obvious what it is.

**Review by chimosky — CHANGES_REQUESTED** [review]

File: cells.py (hunk 13 above)
Comment (Dimi20cen) [reply cells.py]: Do you mean the comments above of the classes? Oh yeah, I agree, fixed

**Pull request merged**